src/logger.ts

```diff
@@ -0,0 +1,48 @@
+import type { CommandInteraction, TextChannel } from "discord.js";
+import type { State } from "./state";
+
+let debugMode = 0;
+
+export function initLogger(initDebugMode: string): void {
+	let parsedDebugMode = parseInt(initDebugMode);
```

Comment:
Perhaps. I was just going for the debugFlag to be toggleable.

I didn't think to have verbosity, would this effect what's displayed in the log channel naturally, or would we still follow only instances where state and interaction are passed through to display?

Reply:
Yeah, the verbosity level (currently called "debug mode") would affect what is posted in the debug channel. Also, I think regardless of the verbosity level, the warn/error message should always be printed in the console.

So maybe rename "debugMode" to "verbosityLevel" that expects a minimum log level (one of: "none", "info", "warn", "error", "debug").

I'm not entirely sure if the interaction and display parameters are necessarily a good idea:
1. The display parameter seems useless, aside from *testing if the logger works* I cannot think of a use case where a message should bypass the specified verbosity level to always be shown.
2. The interaction parameter limits usefulness. For errors not caused as a result of a slash command (e.g., a Mongo error, failure to delete an off-topic #showcase message, etc.), these errors would not have an associated interaction object and thus would not be printed in the debug channel, though they probably should be.

Additionally, I have been considering to create a store for the State that way the state does not have to be explicitly passed to every function that might need it, but rather functions that need it can call `getState()`. But this is better left for another PR.